src/index.ts

```diff
@@ -102,11 +110,17 @@ export default plugin<ThemeFoldOptions>('postcss-theme-fold', (options = { theme
             }
 
             for (const variable of variables) {
+              // Variable absence in uniqVariables means that
+              // it's not present in any theme.
+              node.processed = uniqVariables.has(variable)
+
+              if (!node.processed) {
+                continue
+              }
```

Comment:
Continue, if it's impossible to find variable's value in any theme (css-rule left as is, NOT removed)